### DISCUSSION TITLE
add mingw/bin to path when you install msys2

### DISCUSSION THREAD

**Discussion by Kreijstal** [ScoopInstaller/Main#4001] [Q&A]
why is this not done automatically?

**Comment by rashil2000** [comment]
There's no folder called 'mingw/bin'

**Reply by Kreijstal** [reply]
Okay, sorry I meant $HOME\scoop\apps\msys2\current\mingw64\bin

**Reply by Kreijstal** [reply]
the reason is that programs compiled on mingw don't run because they cannot find their dlls

**Reply by rashil2000** [reply]
Programs compiled with mingw will only work if they are present in the respective bin directory, so that they can find the DLLs. If you copy the programs somewhere else, you need to copy those DLLs also.


Adding all possible (mingw64, mingw32, clang32, clang64, msys, clangarm64) bin directories to PATH will create immense conflicts and is not recommended.

**Reply by Kreijstal** [reply]
do you know how to automatically copy the dlls made from mingw?

**Reply by rashil2000** [reply]
No. But it's easy to do it manually. Just open the program, it will show errors about all missing DLLs, you can copy the DLLs with those names.

**Reply by rashil2000** [reply]
Or better, while compiling your programs, use the `CFLAGS="-static"` CLI flag. Then you won't need any DLLs.
